### PR TITLE
Added a facebook login button using facebook API

### PR DESCRIPTION
**Pull request by stealthanthrax**
Partial fix for issue #935 

Changes: Added login feature using a facebook package

Demo Link: cool-hydrant.surge.sh

Screenshots for the change: 
screen shot 2017-12-11 at 7 17 29 pm

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,6 +51,7 @@
     "react-text-highlight": "^0.2.0",
     "react-textarea-autosize": "^5.1.0",
     "react-url-query": "^1.1.4",
+    "redux": "^3.7.2",
```
Comment (anshumanv) [inline package.json]: What is redux needed for here?

File: public/index.html [hunk 2]
```diff
@@ -38,5 +39,7 @@
       To begin the development, run `npm start`.
       To create a production bundle, use `npm run build`.
     -->
+<script src="https://cdnjs.cloudflare.com/ajax/libs/tether/1.4.0/js/tether.min.js" integrity="sha384-DztdAPBWPRXSA/3eYEEUWrWCy7G5KFbe8fFjk5JAIxUYHKkDx6Qin1DkWx51bBrb" crossorigin="anonymous"></script>
```
Comment (anshumanv) [inline public/index.html]: Indentation

File: src/components/Auth/Login/Login.react.js [hunk 3]
```diff
@@ -118,7 +119,7 @@ class Login extends Component {
 	handleForgotPassword = () => {
 		this.props.handleForgotPassword();
 	}
-
+
```
Comment (anshumanv) [inline src/components/Auth/Login/Login.react.js]: Remove this additional space

File: src/components/Auth/Login/Login.react.js [hunk 4]
```diff
@@ -223,7 +224,12 @@ class Login extends Component {
 			});
 		}
 	}
-
+    
+responseFacebook = (response)=> {
```
Comment (anshumanv) [inline src/components/Auth/Login/Login.react.js]: This function is not needed, I think?

File: src/components/Auth/Login/Login.react.js [hunk 5]
```diff
@@ -311,6 +317,18 @@ class Login extends Component {
 									labelColor="#fff" />
 							</Link>
 						</div>
+						<h4 style={{
+							margin: '8px 0'
+						}}><Translate text="OR"/></h4>
+                                <FacebookLogin socialId="2002829926655326"
```
Comment (anshumanv) [inline src/components/Auth/Login/Login.react.js]: Indentation of this block

Comment (anshumanv) [inline src/components/Auth/Login/Login.react.js]: Is this socialId same for everyone?

**Comment by anshumanv** [comment]
I think you have not added proper oAuth instructions on the fb app, please can you try logging in from the link you gave? thanks :smile:

File: package.json (hunk 1 above)
Comment (stealthanthrax) [reply package.json]: @anshumanv , redux was required for the package to be installed

File: src/components/Auth/Login/Login.react.js (hunk 5 above)
Comment (stealthanthrax) [reply src/components/Auth/Login/Login.react.js]: @anshumanv , this socialId was used just for the purpose of testing,another id through fossasia's credentials should be used

**Comment by stealthanthrax** [comment]
@anshumanv , it was configured to work on localhost:3000, but I will change it now 😄

File: src/components/Auth/Login/Login.react.js (hunk 5 above)
Comment (anshumanv) [reply src/components/Auth/Login/Login.react.js]: Okay, so fossasia's acc need to create an OAuth app and use their ID. :+1:

**Comment by anshumanv** [comment]
Thanks @stealthanthrax for quick changes. :smile:

File: package.json (hunk 1 above)
Comment (anshumanv) [reply package.json]: Are you sure? 
I'm just looking at the component's repo - https://github.com/kennetpostigo/react-facebook-login-component
I didn't find them stating redux as dependency.
Can you please confirm @stealthanthrax, thanks :smile:

**Comment by stealthanthrax** [comment]
screen shot 2017-12-12 at 4 31 38 pm
I get this error @anshumanv

**Comment by anshumanv** [comment]
Hey @stealthanthrax, I cloned your fork and removed the redux dependencies and it ran well on my device, what exactly did you do to remove them?

**Comment by stealthanthrax** [comment]
@anshumanv , to remove them i just "npm install redux@5","npm install react-redux@3","npm install react-container-dimensions@1.x"

**Comment by anshumanv** [comment]
That's to install, please run - 
```sh
$ npm remove redux react-redux -S
$ rm -r node_modules
$ npm install
```
This will remove redux deps package.json and node_modules, remove node_modules, install them again. Though I think only first command should suffice.

**Comment by stealthanthrax** [comment]
@anshumanv , by writing to "remove them" I was referring to removing the unmet peer dependencies

**Comment by stealthanthrax** [comment]
@rishiraj824 , for some reason codacy is not even completing the check and I can't understand why?

**Comment by stealthanthrax** [comment]
@rishiraj824 , I integrated codacy in my personal fork and it is showing this : 
screen shot 2017-12-14 at 4 58 26 pm

File: public/index.html [hunk 6]
```diff
@@ -5,6 +5,7 @@
     <meta content='width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0' name='viewport' />
     <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico">
     <link rel="stylesheet" href="http://cdn.leafletjs.com/leaflet/v0.7.7/leaflet.css" />
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-alpha.6/css/bootstrap.min.css" integrity="sha384-rwoIResjU2yc3z8GV/NPeZWAv56rSmLldC3R/AZzGRnGxQQKnKkoFVhFQhNUwEyJ" crossorigin="anonymous">
```
Comment (madhavrathi) [inline public/index.html]: Why are you using bootstrap? We are already using material UI library.

File: src/components/Auth/Login/Login.react.js [hunk 7]
```diff
@@ -311,6 +313,18 @@ class Login extends Component {
...
+							margin: '8px 0'
+						}}><Translate text="OR"/></h4>
+                    <FacebookLogin socialId="392360367885660"
+                       language="en_US"
+                       scope="public_profile,email"
+                       responseHandler={this.responseFacebook}
+                       xfbml={true}
+                       fields="id,email,name"
+                       version="v2.5"
+                       className="facebook-login btn btn-primary"
```
Comment (madhavrathi) [inline src/components/Auth/Login/Login.react.js]: Please remove bootstrap classes and import FB icon from material UI library. See how other icons are being imported for reference. The icon can be found here: https://www.iconfinder.com/
Make sure you are using free icons.

**Comment by stealthanthrax** [comment]
@madhavrathi ,please have a look

File: public/index.html [hunk 8]
```diff
@@ -38,5 +38,6 @@
       To begin the development, run `npm start`.
       To create a production bundle, use `npm run build`.
     -->
+<script src="https://cdnjs.cloudflare.com/ajax/libs/tether/1.4.0/js/tether.min.js" integrity="sha384-DztdAPBWPRXSA/3eYEEUWrWCy7G5KFbe8fFjk5JAIxUYHKkDx6Qin1DkWx51bBrb" crossorigin="anonymous"></script>
```
Comment (akshatnitd) [inline public/index.html]: Is this required ?

File: package.json [hunk 9]
```diff
@@ -26,10 +26,13 @@
     "react": "^15.5.4",
     "react-bootstrap": "^0.31.0",
     "react-color": "^2.12.1",
+    "react-container-dimensions": "^1.3.3",
```
Comment (akshatnitd) [inline package.json]: You haven't used this package anywhere.

File: package.json [hunk 10]
```diff
@@ -26,10 +26,13 @@
...
     "react-cookie": "^2.0.7",
     "react-custom-scrollbars": "^4.1.2",
     "react-dom": "^15.5.4",
     "react-emojione": "^3.1.10",
+    "react-facebook-login": "^3.7.2",
+    "react-facebook-login-component": "^0.9.1",
```
Comment (akshatnitd) [inline package.json]: You haven't used this package anywhere.

File: package.json [hunk 11]
```diff
@@ -38,6 +41,7 @@
     "react-linkify": "^0.2.1",
     "react-loading-animation": "^1.3.0",
     "react-modal": "^2.2.2",
+    "react-redux": "^5.0.6",
```
Comment (akshatnitd) [inline package.json]: You haven't used this package anywhere.

File: package.json (hunk 9 above)
Comment (stealthanthrax) [reply package.json]: @akshatnitd , this package was required for react-facebook-login , other wise it was showing unmet dependencies

File: package.json (hunk 11 above)
Comment (stealthanthrax) [reply package.json]: so , is this package

File: public/index.html (hunk 8 above)
Comment (stealthanthrax) [reply public/index.html]: @akshatnitd , I forgot to remove this line when I removed bootstrap , I've removed it now,have a look

File: package.json (hunk 10 above)
Comment (akshatnitd) [reply package.json]: @stealthanthrax You haven't used ```react-facebook-login-component```

**Comment by mariobehling** [comment]
A Facebook login should work similarly to the sign up with Facebook on e.g. Pinterest. This needs to be integrated with the server. Simply implementing it without support on the server does not make sense. Therefore I have to close this here. Please help to make the integration.

**Pull request closed without merging**